plugins/modules/ndo_port_channel_interface.py

```diff
@@ -0,0 +1,372 @@
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["name", "uuid"], True],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    node = module.params.get("node")
+    interfaces = module.params.get("interfaces")
+    if interfaces:
+        interfaces = ",".join(interfaces)
+    interface_policy_group = module.params.get("interface_policy_group")
+    interface_policy_group_uuid = module.params.get("interface_policy_group_uuid")
+    interface_descriptions = module.params.get("interface_descriptions")
+    state = module.params.get("state")
+
+    mso_template = MSOTemplate(mso, "fabric_resource", template)
+    mso_template.validate_template("fabricResource")
+    object_description = "Port Channel Interface"
+
+    path = "/fabricResourceTemplate/template/portChannels"
+    existing_port_channel_interfaces = mso_template.template.get("fabricResourceTemplate", {}).get("template", {}).get("portChannels", [])
+
+    if state in ["query", "absent"] and existing_port_channel_interfaces == []:
+        mso.exit_json()
+    elif state == "query" and not (name or uuid):
+        mso.existing = existing_port_channel_interfaces
+    elif existing_port_channel_interfaces and (name or uuid):
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_port_channel_interfaces,
+            [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+
+    ops = []
+
+    if state == "present":
+        if uuid and not mso.existing:
+            mso.fail_json(msg="{0} with the UUID: '{1}' not found".format(object_description, uuid))
+
+        if interface_policy_group and not interface_policy_group_uuid:
+            fabric_policy_template = MSOTemplate(mso, "fabric_policy", interface_policy_group.get("template"))
+            fabric_policy_template.validate_template("fabricPolicy")
+            interface_policy_group_uuid = fabric_policy_template.get_interface_policy_group_uuid(interface_policy_group.get("name"))
+
+        if mso.existing:
+            proposed_payload = copy.deepcopy(match.details)
+
+            if name and mso.existing.get("name") != name:
+                ops.append(dict(op="replace", path="{0}/{1}/name".format(path, match.index), value=name))
+                proposed_payload["name"] = name
+
+            if description is not None and mso.existing.get("description") != description:
```

Comment:
@sajagana  wrote a function for this, asked him to do a separate PR for it so we can rebase the open PRs on it. Shoudl we do this?
https://github.com/CiscoDevNet/ansible-mso/pull/563/files#diff-f3e3720de5c754cfeb714aa1ed0f9d0a1a043f8c0005944d58565decefe1817bR31

Reply:
yeah, let's do it like this. will rebase on this PR once merged.